view/frontend/templates/button_product_page.phtml

```diff
@@ -290,22 +290,34 @@ foreach ($block->getAdditionalCheckoutButtonAttributes() as $attrName => $attrVa
                     <?php foreach ($block->getGroupedProductChildren() as $childProduct):?>
                         var childProductQty = getGroupedProductChildQty(<?= /* @noEscape */ $childProduct->getId(); ?>);
                         if (childProductQty > 0) {
+                            <?php if ($childProduct->isVirtual($childProduct)): ?>
+                                var productType = 'digital';
+                            <?php else: ?>
+                                var productType = 'physical';
+                            <?php endif; ?>
                             items.push({
                                 reference: '<?= /* @noEscape */ $childProduct->getId(); ?>',
                                 price: '<?= /* @noEscape */ $childProduct->getPrice() ?>',
                                 name: '<?= /* @noEscape */ addcslashes($childProduct->getName(), "'"); ?>',
                                 quantity: childProductQty,
-                                options: JSON.stringify(options)
+                                options: JSON.stringify(options),
+                                type: productType
                             });
                         }
                     <?php endforeach; ?>
                 <?php else: ?>
+                    <?php if ($block->getProduct()->isVirtual($block->getProduct())): ?>
```

Comment:
Just to better understand underlying magento code, why do we need `$product->isVirtual($product)` and not `$product->isVirtual()`

Reply:
In this case i've just missed to removed `$product` parameter. 
It won't break anything, but it's unneeded things in the code.
The code that was here before - `$product->getTypeInstance()->isVirtual($product)` - is how it's actually implemented under the hood of isVirtual() function. But obviously we don't need that here as we have just a simple isVirtual() wrapper. 

It's removed with last commit.